oauth2_http/java/com/google/auth/mtls/X509Provider.java

```diff
@@ -135,19 +130,19 @@ public KeyStore getKeyStore() throws IOException {
     } catch (Exception e) {
       // Wrap all other exception types to an IOException.
       throw new IOException(e);
-    } finally {
-      if (certStream != null) {
-        certStream.close();
-      }
-      if (privateKeyStream != null) {
-        privateKeyStream.close();
-      }
-      if (certAndPrivateKeyStream != null) {
-        certAndPrivateKeyStream.close();
-      }
     }
   }
 
+  @Override
+  public boolean isCertificateSourceAvailable() throws IOException {
+    try {
+      this.getKeyStore();
+    } catch (CertificateSourceUnavailableException e) {
+      return false;
+    }
```

Comment:
qq, should this also catch IO exception as well so that it falls back to SecureConnect? Or does it only fallback on CertificateSourceUnavailableException

Reply:
If the design decision is made to have IOExceptions be throw back, can we do two additional things:
1. Make the IOExceptions clearer as to where they stem from (SecureConnect vs X.509) to make this slightly more clear to users? i.e. the IOException message states something like `SecureConnect: Unable/ Error/ etc`. I'm worried that the slight fallback may not be obvious to users (javadocs certainly do help!)
2. Can you add this explanation as a small comment for not catching IOExceptions in `DefaultMtlsProviderFactory.create()` for future maintainers?